WordPress/gutenberg
# Fix BlockInfo slot displaying logic

## What?
Follow up of: https://github.com/WordPress/gutenberg/pull/49819

In the above PR a new private BlockInfo slot was introduced but the conditional display of it's contents was missed. The result was the `add new post` link to be shown in every block:

![screenshot_2023-04-24_at_13 42 02](https://user-images.githubusercontent.com/16275880/234208935-5e62a3ef-b527-4bc9-bf6e-17d189410cf0.png)


 This PR adds the conditional display logic and also removes some related leftovers that weren't used right now.




## Testing Instructions
1. Insert a Query Loop
2. Observe that the `add new post` link is added only in Query Loop and when is selected.
